Examples/Todos/Todos/Todos.swift

```diff
@@ -142,15 +142,40 @@ struct AppView: View {
           Button("Add Todo") { self.viewStore.send(.addTodoButtonTapped, animation: .default) }
         }
       )
-      .environment(
-        \.editMode,
-        self.viewStore.binding(get: \.editMode, send: AppAction.editModeChanged)
-      )
+      .environment(\.editMode, self.viewStore.$editMode)
     }
     .navigationViewStyle(StackNavigationViewStyle())
   }
 }
 
+extension AppState {
+  var view: AppView.ViewState {
+    get { .init(state: self) }
+    set {
+      // handle bindable actions only:
+      self.filter = newValue.filter
+      self.editMode = newValue.editMode
+    }
+  }
+}
+
+extension AppAction {
+  init(action: AppView.ViewAction) {
+    switch action {
+    case .addTodoButtonTapped:
+      self = .addTodoButtonTapped
+    case let .binding(bindingAction):
+      self = .binding(bindingAction.pullback(\AppState.view))
+    case .clearCompletedButtonTapped:
+      self = .clearCompletedButtonTapped
+    case let .delete(indexSet):
+      self = .delete(indexSet)
+    case let .move(source, destination):
+      self = .move(source, destination)
+    }
+  }
+}
```

Comment:
You know what, I completely scanned over this bit when reviewing earlier. Let us think on it a bit more. View state throws a bit of a wrench into something that's otherwise simple...I think we'd probably want to either forego view state entirely in the Todos app (it's simple enough that the scoping we're doing is doubtfully having much of an affect on performance), or maybe avoid introducing binding helpers here... 😔

Reply:
hahahah abort mission!!! All good. I'll close this for now. There's always the option to introduce a seperate example using pullback if needed. Thanks for the feedback.

Reply:
Thanks again for exploring! Hopefully we come up with a better way to work with TCA bindings and view domains in the future that eliminates the extra boilerplate.